jupyterhub/tests/test_pages.py

```diff
@@ -667,3 +667,20 @@ def test_server_not_running_api_request(app):
     assert r.status_code == 404
     assert r.headers["content-type"] == "application/json"
     assert r.json() == {"message": "bees is not running"}
+
+@pytest.mark.gen_test
+def test_metrics_no_auth(app):
+    r = yield get_page("metrics", app)
+    r.raise_for_status()
```

Comment:
Almost there! Instead of calling `raise_for_status()`, we should *expect* a 403 error here:
```suggestion
    assert r.status_code == 403
```

And then remove the url checks below, since it won't send folks to the login page.